lib/internal/mime.js

```diff
@@ -0,0 +1,339 @@
+  MapIteratorPrototypeNext,
+  RegExpPrototypeExec,
+  RegExpPrototypeSymbolMatch,
+  RegExpPrototypeSymbolMatchAll,
+  RegExpPrototypeSymbolReplace,
+  RegExpPrototypeSymbolSearch,
+  RegExpPrototypeSymbolSplit,
+  StringPrototypeReplace,
+  StringPrototypeToLowerCase,
+  StringPrototypeSlice,
+  StringPrototypeCharAt,
+  StringPrototypeIndexOf,
+  StringPrototypeSearch,
+  SymbolIterator,
+  SymbolMatch,
+  SymbolMatchAll,
+  SymbolReplace,
+  SymbolSearch,
+  SymbolSplit,
+  SafeMap: Map,
+} = primordials;
+
+function hardenRegExp(pattern) {
+  pattern[SymbolMatch] = RegExpPrototypeSymbolMatch;
+  pattern[SymbolMatchAll] = RegExpPrototypeSymbolMatchAll;
+  pattern[SymbolReplace] = RegExpPrototypeSymbolReplace;
+  pattern[SymbolSearch] = RegExpPrototypeSymbolSearch;
+  pattern[SymbolSplit] = RegExpPrototypeSymbolSplit;
+  return pattern;
+}
+
+const NotHTTPTokenCodePoint = hardenRegExp(/[^!#$%&'*+\-.^_`|~A-Za-z0-9]/g);
+const NotHTTPQuotedStringCodePoint = hardenRegExp(/[^\t\u0020-~\u0080-\u00FF]/g);
+
+const HTTP_TAB_OR_SPACE = '\t ';
+const HTTP_WHITESPACE = ArrayPrototypeJoin(['\r\n', ...HTTP_TAB_OR_SPACE], '');
+const BEGINNING_WHITESPACE = hardenRegExp(new RegExp(
+  `^[${StringPrototypeReplace(HTTP_WHITESPACE, /\W/g, '\\$&')}]*`));
+const ENDING_WHITESPACE = hardenRegExp(new RegExp(
+  `[${StringPrototypeReplace(HTTP_WHITESPACE, /\W/g, '\\$&')}]*$`));
+
+const ASCII_LOWER = hardenRegExp(/[A-Z]/g);
+function toASCIILower(str) {
+  return StringPrototypeReplace(str, ASCII_LOWER, (c) => StringPrototypeToLowerCase(c));
+}
+
+const SOLIDUS = '/';
+const SEMICOLON = ';';
+function parseTypeAndSubtype(str) {
+  // Skip only HTTP whitespace from start
+  str = StringPrototypeReplace(str, BEGINNING_WHITESPACE, '');
+  // read until '/'
+  const typeEnd = StringPrototypeIndexOf(str, SOLIDUS);
+  const trimmedType = typeEnd === -1 ?
+    str :
+    StringPrototypeSlice(str, 0, typeEnd);
+  if (trimmedType === '' ||
+    StringPrototypeSearch(trimmedType, NotHTTPTokenCodePoint) !== -1) {
+    throw new TypeError('Invalid MIME type');
+  }
+  // skip type and '/'
+  let position = trimmedType.length + 1;
+  const type = toASCIILower(trimmedType);
+  // read until ';'
+  const subtypeEnd = StringPrototypeIndexOf(str, SEMICOLON, position);
+  const rawSubtype = subtypeEnd === -1 ?
+    StringPrototypeSlice(str, position) :
+    StringPrototypeSlice(str, position, subtypeEnd);
+  position += rawSubtype.length;
+  if (subtypeEnd !== -1) {
+    position += 1;
+  }
+  const trimmedSubtype = StringPrototypeReplace(
+    rawSubtype,
+    ENDING_WHITESPACE,
+    '');
+  if (trimmedSubtype === '' ||
+    StringPrototypeSearch(trimmedSubtype, NotHTTPTokenCodePoint) !== -1) {
+    throw new TypeError('Invalid MIME subtype');
+  }
+  const subtype = toASCIILower(trimmedSubtype);
+  const target = ObjectCreate(null);
+  target.type = type;
+  target.subtype = subtype;
+  target.parametersString = StringPrototypeSlice(str, position);
+  return target;
```

Comment:
For this instance it is probably ok, but generally I inherit from `null` unless I have good reason not to as it will avoid some accidents like string coercion. The previous nit to use `Object.create(null)` made it look awkward.

Reply:
What about `__proto__: null`? (Yes I know it's an Annex B feature.)

Reply:
@timothygu [it isn't going to be annex b forever](https://github.com/tc39/ecma262/issues/1595)! The important bit here is it can never be removed so it won't break. Hermes and XS do support this feature already even though they do not completely implement Annex B. It appears Engine262 does not support this, but that seems fine as the other VMs do.

Reply:
https://github.com/nodejs/node/pull/21128#discussion_r351639555 relevant comment requesting me to use Object.create